src/codemagic_cli_tools/apple/app_store_connect/api_session.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING
+
+import requests
+
+from .api_error import AppStoreConnectApiError
+
+if TYPE_CHECKING:
+    from codemagic_cli_tools.apple import AppStoreConnectApiClient
+
+
+class AppStoreConnectApiSession(requests.Session):
+
+    def __init__(self, app_store_connect_api: AppStoreConnectApiClient):
```

Comment:
IMO it would be better to avoid circular dependency like this, at least for the whole class. Would it be possible to pass the headers to the session more explicitly, e.q. by passing a factory function or object?

Reply:
Resolved in 509d040.